engine/consensus/compliance/engine.go

```diff
@@ -149,9 +173,108 @@ func (e *Engine) ProcessLocal(event interface{}) error {
 // Process processes the given event from the node with the given origin ID in
 // a blocking manner. It returns the potential processing error when done.
 func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
-	return e.unit.Do(func() error {
-		return e.process(originID, event)
-	})
+	e.pendingEventSink <- &Event{
+		OriginID: originID,
+		Msg:      event,
+	}
+	return nil
+}
+
+// processEvents is processor of pending events which drives events from networking layer to business logic in `Core`.
+// Effectively consumes messages from networking layer and dispatches them into corresponding sinks which are connected with `Core`.
+// Should be run as a separate goroutine.
+func (e *Engine) processEvents() {
+	// takes pending event from one of the queues
+	// nil sink means nothing to send, this prevents blocking on select
+	fetchEvent := func() (*Event, EventSink, *fifoqueue.FifoQueue) {
+		if val, ok := e.pendingBlocks.Front(); ok {
```

Comment:
Would this give a priority to process pending blocks over pending votes? 

In other words, a node will not process any pending vote unless it has processed all the pending blocks?

Reply:
It will.

Reply:
see my [comment above](https://github.com/onflow/flow-go/pull/406#discussion_r583966779) for an explanation why I think this is currently preferred